hibernate-core/src/main/java/org/hibernate/boot/model/source/spi/PluralAttributeMapKeySource.java

```diff
@@ -39,7 +35,7 @@
 	 * (relevant only for one-to-many and many-to-many associations)?
 	 *
 	 * If this method returns {@code true}, then this object can safely
-	 * be cast to {@link PluralAttributeMapKeySourceEntityAttribute}.
+	 * be cast to {@link PluralAttributeMapKeyManyToManySource}.
```

Comment:
This class is one of the most difficult ones to resolve the Javadoc warnings for semantic understanding is required. Let me know whether my understanding is correct or not. I would not be surprised if my understanding is wrong.